go/processor/floria/processor.go

```diff
@@ -70,31 +73,74 @@ func (p *processor) Run(
 		// Create new contract
 	} else {
 		// Call existing contract
-		result, err = call(p.interpreter, transaction, context)
+		result, err = call(p.interpreter, transaction, context, gas)
 		if err != nil {
-			return errorReceipt, nil
+			return tosca.Receipt{GasUsed: transaction.GasLimit}, nil
 		}
 	}
 
+	gasUsed := gasUsed(transaction, result.GasLeft)
+
 	return tosca.Receipt{
 		Success:         result.Success,
-		GasUsed:         transaction.GasLimit,
+		GasUsed:         gasUsed,
 		ContractAddress: nil,
 		Output:          result.Output,
 		Logs:            nil,
 	}, nil
 }
 
+func gasUsed(transaction tosca.Transaction, gasLeft tosca.Gas) tosca.Gas {
+	if transaction.Sender != (tosca.Address{}) {
+		gasLeft -= gasLeft / 10
+	}
+
+	return transaction.GasLimit - tosca.Gas(gasLeft)
```

Comment:
could `GasLimit` ever be less than `gasLeft` ? should check for underflow ?

Reply:
Not in this spot, gas checks are performed before the execution.